File: auth-data/src/main/java/com/cartobucket/auth/data/domain/Page.java

package com.cartobucket.auth.data.domain;

public record Page(Integer limit, Integer offset) {
    public Page {
        if (limit <= 0)
            throw new IllegalArgumentException("Page limit must be > 0 : " + limit);
        if (limit > 100)
            throw new IllegalArgumentException("Page limit must be <= 100 : " + limit);
        if (offset < 0)
            throw new IllegalArgumentException("Page offset must be >= 0 : " + offset);
    }

    public int getNextRowsCount() {
        return (offset + limit) - 1;
    }
}

File: auth-data/src/main/kotlin/com/cartobucket/auth/data/domain/Page.kt
package com.cartobucket.auth.data.domain

data class Page(val limit: Int, val offset: Int) {
    init {
        require(limit > 0) { "Page limit must be > 0 : $limit" }
        require(limit <= 100) { "Page limit must be <= 100 : $limit" }
        require(offset >= 0) { "Page offset must be >= 0 : $offset" }
    }

    fun getNextRowsCount(): Int = (offset + limit) - 1
    fun offset(): Int {
        return offset
    }
    fun limit(): Int {
        return limit
    }
}